cli/src/clojure_lsp/server.clj

```diff
@@ -219,53 +457,57 @@
+        (logger/debug trace)
+        (recur))))
+  (async/go-loop []
+    (when-let [log-args (async/<! log-ch)]
+      (apply log-wrapper-fn log-args)
+      (recur))))
 
 (defn run-server! []
-  (let [producer* (atom nil)
-        db* db/db*
-        timbre-logger (doto (->TimbreLogger db*)
-                        (logger/setup))
-        _ (logger/info "[SERVER]" "Starting server...")
-        _ (reset! components* (components/->components db* timbre-logger nil))
-        clojure-feature-handler (handlers/->ClojureLSPFeatureHandler components*)
-        server (ClojureLspServer. (LSPServer. clojure-feature-handler
-                                              producer*
-                                              db*
-                                              db/initial-db
-                                              capabilites
-                                              client-settings
-                                              known-files-pattern)
-                                  clojure-feature-handler)
-        in System/in
-        out System/out
-        ;; For debugging, it's possible to trace all I/O through lsp4j.
-        ;; launcher (Launcher/createLauncher server ClojureLanguageClient in out false (java.io.PrintWriter. (io/writer (io/file "../../integration-test/sample-test/clojure-lsp.lsp-trace.out"))))
-        launcher (Launcher/createLauncher server ClojureLanguageClient in out)
-        language-client ^ClojureLanguageClient (.getRemoteProxy launcher)
-        producer (->ClojureLspProducer language-client
-                                       (lsp/->LSPProducer language-client db*)
-                                       db*)]
-    ;; TODO remove atom, think in a way to build all components in the same place and not need to assoc to atom later.
-    (reset! producer* producer)
-    (swap! components* assoc :producer producer)
-    (nrepl/setup-nrepl db*)
-    (spawn-async-tasks! @components*)
-    (.startListening launcher)))
+  (lsp.server/discarding-stdout
+    (let [timbre-logger (->TimbreLogger)
+          log-path (logger/setup timbre-logger)
+          db (assoc db/initial-db :log-path log-path)
+          db* (atom db)
+          server (lsp.server/stdio-server {:trace? false
+                                           :in System/in
+                                           :out System/out})
+          producer (ClojureLspProducer. server db*)
+          components {:db* db*
+                      :logger timbre-logger
+                      :producer producer
+                      :server server}]
+      (logger/info "[SERVER]" "Starting server...")
+      ;; TODO: if this were moved to `initialize`, after timbre has been
+      ;; configured, the server's startup logs and traces would appear in the
+      ;; regular log file instead of the temp log file. The downside would be that
+      ;; if anything bad happens before `initialize`, we wouldn't get any logs.
```

Comment:
Yes, it happened before exceptions before we update the new log path, so makes sense